src/btree/bt_npos.c

```diff
@@ -0,0 +1,352 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+
+/*
+ * !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+ *
+ * Below is a set of functions for computing the normalized position of a page and restoring a page
+ * from its normalized position. These functions are used by the eviction server for the sake of
+ * not holding the hazard pointer for longer than necessary. Another user is "partition cursor".
+ *
+ * Normalized position is a number in the range of 0 .. 1 that represents a page's position across
+ * all pages. It's primary goal is to be cheap rather than precise. It works best when the
+ * tree is perfectly balanced, i.e. all internal pages at the same level have the same number of
+ * children and the depth of all leaf pages is the same. In practice, the tree is not perfect, so
+ * the normalized position is imprecise. However, it's totally fine for the eviction server because
+ * it only uses an approximate position in the tree to continue to walk from.
+ * Even when using a hazard pointer, page splits can shift data so that some pages or sub-trees can
+ * be skipped an eviction pass.
+ *
+ * Eviction wants to be as non-intrusive as possible and never loads pages into memory, while
+ * Partition Cursor can load pages or wait for them to be unlocked. The behavior is controlled by
```

Comment:
I don't understand the reference to "Partition Cursor" here. Also the word "a" should come before it.

Reply:
I made it plural.
Partition cursor is an upcoming project with Server. It doesn't exist yet but I don't think this block of description is the right place to elaborate on it.